mlflow/sklearn/__init__.py

```diff
@@ -455,6 +460,323 @@ def load_model(model_uri):
     )
 
 
+class _AutologTrainingStatus:
+    """
+    This class is designed for holding information which is used by autologging metrics
+    It will hold information of:
+    (1) a map of "prediction result object id" to a tuple of dataset name(the dataset is
+       the one which generate the prediction result) and run_id.
```

Comment:
Can we add an explanation to this docstring for why we need this map instead of just setting the `run_id` attribute on the prediction result? I.e. because certain dataset types, like numpy arrays, don't support attribute assignment.